src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,162 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> first;
+    private Node<T> last;
+    private int size;
+
     @Override
     public boolean add(T value) {
-        return false;
+        Node<T> newNode = new Node<>(last, value, null);
+        if (last == null) {
+            first = newNode;
+        } else {
+            last.next = newNode;
+        }
+        last = newNode;
+        size++;
+        return true;
     }
 
     @Override
     public void add(T value, int index) {
+        Node<T> currentNode = findElement(index);
+        checkIndex(index, index > size);
+        if (index == 0) {
+            addFirst(value, currentNode);
+            return;
+        }
+        if (index == size) {
+            add(value);
+        } else {
+            Node<T> prev = currentNode.prev;
+            Node<T> newNode = new Node<>(prev, value, currentNode);
+            currentNode.prev = newNode;
+            if (prev == null) {
+                first = newNode;
+            } else {
+                prev.next = newNode;
+            }
+            size++;
+        }
+    }
 
+    private void addFirst(T value, Node<T> currentNode) {
+        Node<T> newNode = new Node<>(null, value, currentNode);
+        first = newNode;
+        if (currentNode == null) {
+            last = newNode;
+        } else {
+            currentNode.prev = newNode;
+        }
+        size++;
     }
 
     @Override
     public boolean addAll(List<T> list) {
-        return false;
+        if (list == null) {
+            return false;
+        }
+        for (T value : list) {
+            add(value);
+        }
+        return true;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index, index >= size);
+        return findElement(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndex(index, index >= size);
+        Node<T> newNode = findElement(index);
+        T oldValue = newNode.item;
+        newNode.item = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index, index >= size);
+        return delete(index);
```

Comment:
be attentive. I wrote from line 87, not 86 - https://github.com/mate-academy/jv-linked-list/pull/207#discussion_r542504193
```suggestion
        return delete(findElement(index));
```

Reply:
done